### PR TITLE
feat(core): improve error message when items cannot be found

### LINKED ISSUE #108: Improve error message when generation could not be found
Currently the error message when a generation could not be found dumps all information about all generations. 

That's too much information. It would be more helpful to just print out the possible names.

### PR DESCRIPTION
**Pull request by pihme**
closes #108

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -86,6 +86,16 @@
 			<version>${version.google-http-client}</version>
 		</dependency>
 
+		<dependency>
+			<groupId>io.vavr</groupId>
+			<artifactId>vavr</artifactId>
```
Comment (Zelldon) [inline core/pom.xml:91]: I have read about this lib before. Did you used it before?

File: core/src/main/java/io/zeebe/clustertestbench/handler/CreateGenerationInCamundaCloudHandler.java [hunk 2]
```diff
@@ -89,34 +87,28 @@ protected static String createGenerationName(String zeebeImage) {
...
 
-    if (optTemplateGenerationInfo.isEmpty()) {
-      throw new RuntimeException("Unable to find generation: " + generationTemplate);
-    }
-    return optTemplateGenerationInfo.get();
+    return eitherTemplateGenerationInfo.getOrElseThrow(msg -> new IllegalArgumentException(msg));
   }
 
-  private Optional<GenerationInfo> findGenerationInfoByName(final String name) {
+  private Either<String, GenerationInfo> findGenerationInfoByName(final String name) {
```
Comment (Zelldon) [inline core/src/main/java/io/zeebe/clustertestbench/handler/CreateGenerationInCamundaCloudHandler.java:95]: Could we replace our either in zeebe with vavr? Probably ?

File: core/src/main/java/io/zeebe/clustertestbench/util/StringLookup.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+   * @param candidates the candidates to be searched
+   * @param extractor function that extracts the relevant string from a candidate
+   * @param ignoreCase flag whether the search should respect or ignore case
+   */
+  public StringLookup(
+      String entityToFind,
+      String searchTerm,
+      List<T> candidates,
+      Function<T, String> extractor,
+      boolean ignoreCase) {
```
Comment (Zelldon) [inline core/src/main/java/io/zeebe/clustertestbench/util/StringLookup.java:53]: You don't need to change it is just a comment:

I'm not a fan of boolean flags, ideally we had something speaking either a enum or classes which indicate the behavior. Like CaseSensitiveLookup and InSensitiveLookup, internally they have a composition of the StringLookup logic and use the different prefixes.

**Review by Zelldon — APPROVED** [review]
Thanks :+1:  I was in the beginning not sure what we want to solve :D But I think I get it now :+1:

File: core/pom.xml (hunk 1 above)
Comment (pihme) [reply core/pom.xml:91]: Not yet. That's why I want to use it here. I find it easier to introduce dependencies and use them intensively in side projects first.

File: core/src/main/java/io/zeebe/clustertestbench/handler/CreateGenerationInCamundaCloudHandler.java (hunk 2 above)
Comment (pihme) [reply core/src/main/java/io/zeebe/clustertestbench/handler/CreateGenerationInCamundaCloudHandler.java:95]: We could, but I as far as I remember Nico and Phil evaluated it in the beginning and decided against it.

**Pull request merged**